src/Lower.cpp

```diff
@@ -271,6 +272,12 @@ Stmt lower(vector<Function> outputs, const string &pipeline_name, const Target &
     s = simplify(s);
     debug(1) << "Lowering after final simplification:\n" << s << "\n\n";
 
+    if (t.has_feature(Target::MSAN)) {
```

Comment:
The .buffer symbols that this relies on may have already been stripped during simplification because they were unused.  I'd move this lowering pass to just after storage flattening.